src/features/MoviesListPage/index.js

```diff
@@ -5,48 +5,41 @@ import { fetchMoviesList, selectImagePath, selectLoading } from '../../core/movi
 import MoviesList from '../MoviesList';
 import { Header } from '../../common/Header/styled';
 import Paginator from '../../common/Paginator';
-import { selectactualPage, selectActualQuery, setActualLocation, setactualPage } from '../../core/actual/actualStateSlice';
 import { Container } from "../../common/Container/styled";
 import { Loading } from "../../common/Message/MessageContainer/Loading";
 import { NoResults } from '../../common/Message/MessageContainer/NoResults';
+import useQueryParameter from '../../core/search/useQueryParameter';
+import searchQueryParameter from '../../core/search/searchQueryParameter';
 
 export const MoviesListPage = () => {
     const dispatch = useDispatch();
     const navigate = useNavigate();
     const location = useLocation();
     const { page: urlPage } = useParams();
-    const page = useSelector(selectactualPage);
     const movies = useSelector(selectImagePath);
     const loading = useSelector(selectLoading);
     const baseURL = `${"https://image.tmdb.org/t/p/"}${"w500"}`;
     const actualPage = parseInt(urlPage, 10) || 1;
```

Comment:
Maybe:
const actualPage = Math.max(1, Math.min(parseInt(page, 10) || 1, lastPage));
This will ensure that actualPage is never less than 1 and never greater than lastPage